application.py

```diff
@@ -110,18 +110,18 @@ def render_game_report(id):
 def exchange_favor(game_id):
     formdata = request.form
     try:
-        giver_id = formdata["giver_id"]
-    except:
-        giver_id = session["id"]
-    if giver_id is None:
-        giver_id = session["id"]
+        giver = formdata["giver_id"]
```

Comment:
not sure if `giver` is better than `giver_id`, to get out of the db: _id suffix is more descriptive. might be upstream limitation?